src/javascripts/ng-admin/Crud/routing.js

```diff
@@ -55,17 +62,83 @@ define(function (require) {
                 controllerAs: 'listController',
                 templateProvider: templateProvider('ListView', listTemplate),
                 resolve: {
+                    dataStore: dataStoreProvider(),
                     view: viewProvider('ListView'),
-                    data: ['$stateParams', 'RetrieveQueries', 'view', function ($stateParams, RetrieveQueries, view) {
+                    response: ['$stateParams', 'RetrieveQueries', 'view', function ($stateParams, RetrieveQueries, view) {
                         var page = $stateParams.page,
                             filters = $stateParams.search,
                             sortField = $stateParams.sortField,
                             sortDir = $stateParams.sortDir;
 
-                        return RetrieveQueries.getAll(view, page, true, filters, sortField, sortDir);
+                        return RetrieveQueries.getAll(view, page, filters, sortField, sortDir);
+                    }],
+                    totalItems: ['response', function (response) {
+                        return response.totalItems;
+                    }],
+                    referencedData: ['RetrieveQueries', 'view', 'response', function (RetrieveQueries, view, response) {
+                        return RetrieveQueries.getReferencedData(view.getReferences(), response.data);
+                    }],
+                    referencedEntries: ['dataStore', 'view', 'referencedData', function (dataStore, view, referencedData) {
+                        var references = view.getReferences();
+                        var referencedEntries;
+
+                        for (var name in referencedData) {
+                            referencedEntries = dataStore.mapEntries(
+                                references[name].targetEntity().name(),
+                                references[name].targetEntity().identifier(),
+                                [references[name].targetField()],
+                                referencedData[name]
+                            );
+
+                            dataStore.setEntries(
+                                references[name].targetEntity().uniqueId + '_values',
+                                referencedEntries
+                            );
+                        }
+
+                        return true;
+                    }],
+                    entries: ['dataStore', 'view', 'response', 'referencedEntries', function (dataStore, view, response, referencedEntries) {
+                        var entries = dataStore.mapEntries(
+                            view.entity.name(),
+                            view.identifier(),
+                            view.getFields(),
+                            response.data
+                        );
+
+                        // shortcut to diplay collection of entry with included referenced values
+                        dataStore.fillReferencesValuesFromCollection(entries, view.getReferences(), true);
```

Comment:
This method is use only one time, but `setEntries` is use in multiple cases, I prefer be explicit, and call it in the routing, what do you think ?